flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/BinaryType.java

```diff
@@ -91,9 +92,11 @@ private BinaryType(int length, boolean isNullable) {
 	 * For consistent behavior, the same logic applies to binary strings.
 	 *
 	 * <p>This method enables this special kind of binary string.
+	 *
+	 * <p>Zero-length binary strings have no serializable string representation.
 	 */
 	public static BinaryType ofEmptyLiteral() {
-		return new BinaryType(EMPTY_LITERAL_LENGTH, false);
+		return new BinaryType(EMPTY_LITERAL_LENGTH, true);
```

Comment:
Is it possible to create a nullable empty literal? I think empty string literals by the contract of our API will always be non null.

Reply:
Agreed. I did this change because by default types in SQL are always nullable, but we don't need to do it for this special case.

Reply:
I prefer `ofEmptyLiteral(boolean isNullable)`.
Literal empty string will be empty literal non-nullable CharType.
But there are some nullable cases too.
Consider `substring('', 1, 3)`, I think we should return an empty literal nullable VarCharType in blink runner, Unlike Flink runner, Blink runner tries not to throw exceptions, which leads to task failure. Blink runner chooses return null to when error in calculation. (be consistent with Hive and Spark)